### PR TITLE
dhcp: T5952: Fix validate duplicate MAC Address on same subnet

### PR DESCRIPTION
**Pull request by josephillips85**
## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T5952

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Review by c-po — APPROVED** [review]

File: src/conf_mode/service_dhcp-server.py [hunk 1]
```diff
@@ -247,7 +248,11 @@ def verify(dhcp):
                         if mapping_config['ip_address'] in used_ips:
                             raise ConfigError(f'Configured IP address for static mapping "{mapping}" exists on another static mapping')
 
+                        if mapping_config['mac'] in used_mac:
+                            raise ConfigError(f'Configured MAC address for static mapping "{mapping}" exists on another static mapping')
+
                         used_ips.append(mapping_config['ip_address'])
+                        used_mac.append(mapping_config['mac'])
```
Comment (sarthurdev) [inline src/conf_mode/service_dhcp-server.py:256]: This needs to be inside a check that the mac field is present. The code can reach this point with only a DUID and IP set.

**Review by sarthurdev — CHANGES_REQUESTED** [review]

File: src/conf_mode/service_dhcp-server.py (hunk 1 above)
Comment (c-po) [reply src/conf_mode/service_dhcp-server.py:256]: Good catch! We should have an identical if statement for the `mac` address key

**Review by c-po — CHANGES_REQUESTED** [review]

File: src/conf_mode/service_dhcp-server.py (hunk 1 above)
Comment (josephillips85) [reply src/conf_mode/service_dhcp-server.py:256]: Ok i perform the corrections the IF statement should't be identical . acording the documentation you are able to set a mac address only and will assigned by the dynamic pool .

Comment (c-po) [reply src/conf_mode/service_dhcp-server.py:256]: I wonder if we should add a check for duplicate duids as well @sarthurdev?

**Review by c-po — APPROVED** [review]

**Pull request merged**